### PR TITLE
feat: Adds useBrowserOffline

### PR DESCRIPTION
**Pull request by edas**
This hook is extracted from cozy-notes, to simplify
the offline indicator there. It should return if the browser
is online (default to 'yes' if we have no indicator, like in
Node.js) and react to online and offline events.

### CODE REVIEW

**Comment by Crash--** [comment]
I don't know if UI is the right place to put this kind of hook? 

It's hard to know if the hook is in UI or in cozyclient or somewhere else... 

I think having this logic in CozyClient makes more sense since offline / inline is not only about displaying UI but can be used also to do cozy client related stuff.

**Comment by ptbrowne** [comment]
Personally, I think it belongs more in cozy-ui since it does not use anything from the client, I see cozy-ui as our general toolbox. This is not blocking for me.

I agree for the need for documentation :).

File: react/hooks/useBrowserOffline.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useState, useCallback } from 'react'
+import useEventListener from './useEventListener'
+
+export default function useBrowserOffline() {
+  function isOfflineNow() {
+    return window && window.navigator && window.navigator.onLine === false
+  }
+
+  const [isOffline, setOfflineSince] = useState(isOfflineNow)
```
Comment (y-lohse) [inline react/hooks/useBrowserOffline.js]:
```suggestion
  const [isOffline, setOfflineSince] = useState(isOfflineNow())
```

… at least I think? But then why didn't the test catch that?

Comment (edas) [reply react/hooks/useBrowserOffline.js]: Yu can pass a function to useState, it will be executed at the first initialization

**Comment by edas** [comment]
> Also, if we keep it in UI, we should add it to the documentation

done

> I don't know if UI is the right place to put this kind of hook?

I have no strong opinion about this. I've used UI because there is no interaction with the server or the data. We already have useEventlistener there.

File: react/hooks/useBrowserOffline.js (hunk 1 above)
Comment (y-lohse) [reply react/hooks/useBrowserOffline.js]: TIL!

**Pull request merged**